services/navigation/builtin/builtin.go

```diff
@@ -140,6 +136,18 @@ func (conf *Config) Validate(path string) ([]string, error) {
 		return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
 	}
 
+	// Ensure we have obstacle detector(s) if MapType is GPS (or default)
+	if mapType == navigation.GPSMap && len(conf.ObstacleDetectors) == 0 {
```

Comment:
I would say that a motion service is necessary for None map as well, as without it the explore mode is available. Waypoint mode is also unavailable due to lack of a map.

Therefore a None map_type with no vision service would only be able to do manual mode, i.e. nothing.

Reply:
I am a little confused by what you mean here.
Regardless of `MapType` we already do 
```
	// Add motion service dependencies
	if conf.MotionServiceName != "" {
		deps = append(deps, resource.NewName(motion.API, conf.MotionServiceName).String())
	} else {
		deps = append(deps, resource.NewName(motion.API, resource.DefaultServiceName).String())
	}
```
Do you mean to say that a vision service and therefore a valid `ObstacleDetector` is needed for a `MapType` of `None`?

Reply:
I don't think a vision service is required  for map type none. We said in the explore scope:

> Add a map_type to the navigation service config. At config validation we will check that the required components and services exist for the given map_type
map_type: “GPS” - movement sensor, base
map_type: “None” - base

Reply:
Apologies for the confusion! 

I remember this conversation, but we also said GPD wouldn't require a vision service and with the changes to validate, it now does

Reply:
I think it makes sense for GPS/default to require a valid `ObstacleDetector` and therefore a vision service.
I see that in `builtin.go` we have
```
availableModesByMapType = map[navigation.MapType][]navigation.Mode{
	navigation.NoMap:  {navigation.ModeManual, navigation.ModeExplore},
	navigation.GPSMap: {navigation.ModeManual, navigation.ModeWaypoint, navigation.ModeExplore},
}
```
My reasoning is that GPS allows `navigation.ModeWaypoint` which calls `MoveOnGlobe`, so we will need to have a vision service and camera we can reference.

Reply:
Map type GPS must not require an ObstacleDetector, since we support waypoint mode without transient obstacle detection.

I think it also makes sense to allow Explore mode without an ObstacleDetector, but I don't feel as strongly about this.